### PR TITLE
Help

### PR DESCRIPTION
**Pull request by zachstence**
Create a basic Help command, used by specifying command documentation in the `@Command` decorator

Its not pretty right now, but I'm going to make some changes to the `@Command` decorator and then catch the Help stuff up afterwards

help
helpspec

### CODE REVIEW

File: src/administrator-commands/sprocket-configuration.marshal.ts [hunk 1]
```diff
@@ -6,7 +6,12 @@ import {
 import {table} from "../util/table";
 
 export class SprocketConfigurationMarshal extends Marshal {
-    @Command(["getSprocketConfig", "getSprocketConfig key"])
+    @Command(["getSprocketConfig", "getSprocketConfig key"], {
```
Comment (0c370t) [inline src/administrator-commands/sprocket-configuration.marshal.ts:9]: Probably out of scope but this needs to be split into 2 commands

File: src/marshal/command-manager/command-manager.service.ts [hunk 2]
```diff
@@ -9,12 +9,12 @@ export class CommandManagerService {
     /**
      * List of all registered commands
      */
-    commands = new Map<string, Required<CommandMetadata>>();
+    private commands = new Map<string, Required<CommandMetadata>>();
```
Comment (0c370t) [inline src/marshal/command-manager/command-manager.service.ts:12]: Why the access change?

File: src/marshal/command-manager/command-manager.service.ts [hunk 3]
```diff
@@ -127,4 +127,8 @@ export class CommandManagerService {
         this.commandNotFoundHooks.add(commandNotFound);
     }
 
+    getCommands(): Map<string, Required<CommandMetadata>> {
```
Comment (0c370t) [inline src/marshal/command-manager/command-manager.service.ts:130]: use a getter?

File: src/member-commands/help.marshal.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+     *              type: String,
+     *              optional: true,
+     *              desc: ...,
+     *          },
+     *      ],
+     *      desc: ...,
+     *      longDesc: ...,
+     *  })
+     * We probably don't need the concept of aliases anymore, check with Sov
+     */
```
Comment (0c370t) [inline src/member-commands/help.marshal.ts:30]: Is comment required here?

File: src/member-commands/help.marshal.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+            }
+
+            await this.helpDetail(m, metadata);
+            return;
+        }
+
+        // If commandName not given, show help for all commands
+        const commandMetas = [...this.cms.getCommands().values()];
+
+        // TODO get organization from author on context?
```
Comment (0c370t) [inline src/member-commands/help.marshal.ts:66]: Will likely be more complicated, might be worth a brainstorming doc

File: src/member-commands/help.marshal.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        const embed = await this.embedService.embed({
+            title: "Help",
+            fields: commandMetas.map(this.metadataToField),
+        });
+
+        await m.reply({embeds: [embed] });
+    }
+
+    private async helpDetail(m: Message, meta: CommandMetadata): Promise<void> {
```
Comment (0c370t) [inline src/member-commands/help.marshal.ts:75]: Service please

File: src/member-commands/help.marshal.ts [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+                    value: "**Arguments**",
+                },
+                ...argsDocs,
+            ],
+        });
+
+        await m.reply({embeds: [embed] });
+    }
+
+    private metadataToField = (meta: CommandMetadata): EmbedFieldData => ({
```
Comment (0c370t) [inline src/member-commands/help.marshal.ts:102]: Service please

File: src/util/config.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+if (!config.has("bot.prefix")) {
+    throw new Error("Missing required config bot.prefix");
+}
+
+export const appConfig = {
+    bot: {
+        prefix: config.get<string>("bot.prefix"),
+    },
+};
```
Comment (0c370t) [inline src/util/config.ts:11]: Make a ticket to do this for all the things

**Review by 0c370t — CHANGES_REQUESTED** [review]

File: src/administrator-commands/sprocket-configuration.marshal.ts (hunk 1 above)
Comment (zachstence) [reply src/administrator-commands/sprocket-configuration.marshal.ts:9]: Yeah will tackle this after #16

File: src/util/config.ts (hunk 8 above)
Comment (zachstence) [reply src/util/config.ts:11]: https://app.clickup.com/t/1zvw33p

File: src/member-commands/help.marshal.ts (hunk 7 above)
Comment (zachstence) [reply src/member-commands/help.marshal.ts]: eed6886

File: src/member-commands/help.marshal.ts (hunk 6 above)
Comment (zachstence) [reply src/member-commands/help.marshal.ts]: 9f8d490

File: src/member-commands/help.marshal.ts (hunk 5 above)
Comment (zachstence) [reply src/member-commands/help.marshal.ts]: https://www.notion.so/sovietshark/Command-Context-67411b44dc154a249eb657f7119876f2

File: src/member-commands/help.marshal.ts (hunk 4 above)
Comment (zachstence) [reply src/member-commands/help.marshal.ts]: 01b74af

File: src/marshal/command-manager/command-manager.service.ts (hunk 3 above)
Comment (zachstence) [reply src/marshal/command-manager/command-manager.service.ts]: 900b03f

File: src/marshal/command-manager/command-manager.service.ts (hunk 2 above)
Comment (zachstence) [reply src/marshal/command-manager/command-manager.service.ts]: 900b03f

**Review by 0c370t — APPROVED** [review]

**Pull request merged**